components/raftstore-v2/src/operation/command/admin/split.rs

```diff
@@ -249,6 +267,175 @@ impl<EK: KvEngine, R: ApplyResReporter> Apply<EK, R> {
     }
 }
 
+impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
+    pub fn on_ready_split_region<T>(
+        &mut self,
+        store_ctx: &mut StoreContext<EK, ER, T>,
+        derived_index: usize,
+        tablet_index: u64,
+        regions: Vec<Region>,
+    ) {
+        fail_point!("on_split", self.peer().get_store_id() == 3, |_| {});
+
+        let derived = &regions[derived_index];
+        let derived_epoch = derived.get_region_epoch().clone();
+        let region_id = derived.get_id();
+
+        // Group in-memory pessimistic locks in the original region into new regions.
+        // The locks of new regions will be put into the corresponding new regions
+        // later. And the locks belonging to the old region will stay in the original
+        // map.
+        let region_locks = {
+            let mut pessimistic_locks = self.txn_ext().pessimistic_locks.write();
+            info!(self.logger, "moving {} locks to new regions", pessimistic_locks.len(););
+            // Update the version so the concurrent reader will fail due to EpochNotMatch
+            // instead of PessimisticLockNotFound.
+            pessimistic_locks.version = derived_epoch.get_version();
+            pessimistic_locks.group_by_regions(&regions, derived)
+        };
+        fail_point!("on_split_invalidate_locks");
+
+        // Roughly estimate the size and keys for new regions.
+        let new_region_count = regions.len() as u64;
+        let mut meta = store_ctx.store_meta.lock().unwrap();
+        meta.set_region(
+            derived.clone(),
+            self,
+            RegionChangeReason::Split,
+            tablet_index,
+        );
+        self.post_split();
```

Comment:
Must it be inside lock? If so add comment in `post_split` to require the lock.

Reply:
No, I will change this.